apps/condo/domains/acquiring/schema/Payment.js

```diff
@@ -139,9 +153,38 @@ const Payment = new GQLListSchema('Payment', {
             },
         },
 
+        invoice: {
+            schemaDoc: 'Link to the invoice that user paid for.',
+            type: 'Relationship',
+            ref: 'Invoice',
+            isRequired: false,
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+            hooks: {
+                validateInput: ({ context, resolvedData, fieldPath }) => {
+                    if (resolvedData[fieldPath] && !resolvedData['frozenInvoice']) {
+                        throw new GQLError(ERRORS.PAYMENT_NO_PAIRED_FROZEN_INVOICE, context)
+                    }
+                },
+            },
+        },
+
+        frozenInvoice: {
+            schemaDoc: 'Frozen invoice, used to resolving conflicts',
+            type: 'Json',
```

Comment:
Could it be strongly typed?

Reply:
For now, it's not needed for this field. All frozen fields will be refactored.